cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java

```diff
@@ -420,6 +428,42 @@ public void testNewCommandCentralPullWithPlatformDependencies() throws IOExcepti
         Assert.assertTrue(readOutput().contains("Created new package"));
     }
 
+    @Test(description = "Test new command by pulling a central template having a central dependency")
+    public void testNewCommandCentralPullWithCentralDependency() throws IOException {
+        // Cache dependency to central --> pramodya/winery:0.1.0
+        cacheBalaToCentralRepository(testResources.resolve("balacache-dependencies").resolve("pramodya")
+                .resolve("winery").resolve("0.1.0").resolve("any"), "pramodya", "winery", "0.1.0", "any");
+
+        // Publish template has dependency
+        // pramodya/template_lib_project:1.0.0 --> pramodya/winery:0.1.0
+        cacheBalaToCentralRepository(testResources.resolve("balacache-dependencies").resolve("pramodya")
+                        .resolve("template_lib_project").resolve("1.0.0").resolve("any"), "pramodya",
+                "template_lib_project", "1.0.0", "any");
+
+        // Cache updated dependency to central --> pramodya/winery:2.1.0
+        cacheBalaToCentralRepository(testResources.resolve("balacache-dependencies").resolve("pramodya")
+                .resolve("winery").resolve("2.1.0").resolve("any"), "pramodya", "winery", "2.1.0", "any");
+
+        // Create a new package using `template_lib_project` template
+        String templateArg = "pramodya/template_lib_project:1.0.0";
+        String packageName = "sample_lib_project";
+        String[] args = {packageName, "-t", templateArg};
+        NewCommand newCommand = new NewCommand(tmpDir, printStream, false, homeCache);
+        new CommandLine(newCommand).parseArgs(args);
+        newCommand.execute();
+
+        Path packageDir = tmpDir.resolve(packageName);
+        Assert.assertTrue(Files.exists(packageDir));
+        Assert.assertTrue(readOutput().contains("Created new package"));
+        Assert.assertTrue(Files.exists(packageDir.resolve(ProjectConstants.BALLERINA_TOML)));
+        String depsTomlContent = Files.readString(
+                packageDir.resolve(ProjectConstants.DEPENDENCIES_TOML), StandardCharsets.UTF_8);
+        Assert.assertTrue(depsTomlContent.contains("[[package]]\n" +
+                "org = \"pramodya\"\n" +
+                "name = \"winery\"\n" +
+                "version = \"0.1.0\""));
```

Comment:
Shall we execute the build command after this to make sure the package is built successfully